x-pack/platform/plugins/shared/alerting/server/task_runner/ad_hoc_task_runner.ts

```diff
@@ -630,4 +638,31 @@ export class AdHocTaskRunner implements CancellableTask {
       );
     }
   }
+
+  private async updateGapsAfterBackfillComplete() {
+    if (!this.shouldDeleteTask) return;
+
+    if (this.scheduleToRunIndex < 0 || !this.adHocRange) return null;
+
+    const fakeRequest = getFakeKibanaRequest(
```

Comment:
there is some overhead in generating a fake request, so instead of saving the apiKeyToUse as a variable, can we just save the fakeRequest that's generated in the `prepareToRun` function and reuse here?